### PR TITLE
Add Collision_distance_field 

### PR DESCRIPTION
**Pull request by anasarrak**

### CODE REVIEW

File: moveit_core/collision_distance_field/CMakeLists.txt [hunk 1]
```diff
@@ -14,28 +14,27 @@ add_library(${MOVEIT_LIB_NAME}
...
+        ARCHIVE DESTINATION lib)
 
-install(TARGETS talker
-  LIBRARY DESTINATION lib
-  RUNTIME DESTINATION bin
-)
+install(DIRECTORY include/ DESTINATION include)
 
-install(DIRECTORY include/ DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION})
+# if(BUILD_TESTING)
```
Comment (ahcorde) [inline moveit_core/collision_distance_field/CMakeLists.txt:33]: I would remove this until we get some tests.

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 2]
```diff
@@ -410,19 +409,21 @@ bool collision_detection::doBoundingSpheresIntersect(const PosedBodySphereDecomp
...
 
 void collision_detection::getCollisionSphereMarkers(
-    const std_msgs::ColorRGBA& color, const std::string& frame_id, const std::string& ns, const ros::Duration& dur,
-    const std::vector<PosedBodySphereDecompositionPtr>& posed_decompositions, visualization_msgs::MarkerArray& arr)
+    const std_msgs::msg::ColorRGBA& color, const std::string& frame_id, const std::string& ns, const rclcpp::Duration& dur,
+    const std::vector<PosedBodySphereDecompositionPtr>& posed_decompositions, visualization_msgs::msg::MarkerArray& arr)
 {
   unsigned int count = 0;
+  rclcpp::Clock ros_clock;
+  rclcpp::Time ros_now = ros_clock.now();
```
Comment (ahcorde) [inline moveit_core/collision_distance_field/src/collision_distance_field_types.cpp]: Remove this line, read comment below

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 3]
```diff
@@ -410,19 +409,21 @@ bool collision_detection::doBoundingSpheresIntersect(const PosedBodySphereDecomp
...
     if (posed_decompositions[i])
     {
       for (unsigned int j = 0; j < posed_decompositions[i]->getCollisionSpheres().size(); j++)
       {
-        visualization_msgs::Marker sphere;
-        sphere.type = visualization_msgs::Marker::SPHERE;
-        sphere.header.stamp = ros::Time::now();
+        visualization_msgs::msg::Marker sphere;
+        sphere.type = visualization_msgs::msg::Marker::SPHERE;
+        sphere.header.stamp = ros_now;
```
Comment (ahcorde) [inline moveit_core/collision_distance_field/src/collision_distance_field_types.cpp]: I will use rewrite this:

```cpp
sphere.header.stamp = ros_clock.now();
```

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 4]
```diff
@@ -440,14 +441,16 @@ void collision_detection::getCollisionSphereMarkers(
...
 void collision_detection::getProximityGradientMarkers(
-    const std::string& frame_id, const std::string& ns, const ros::Duration& dur,
+    const std::string& frame_id, const std::string& ns, const rclcpp::Duration& dur,
     const std::vector<PosedBodySphereDecompositionPtr>& posed_decompositions,
     const std::vector<PosedBodySphereDecompositionVectorPtr>& posed_vector_decompositions,
-    const std::vector<GradientInfo>& gradients, visualization_msgs::MarkerArray& arr)
+    const std::vector<GradientInfo>& gradients, visualization_msgs::msg::MarkerArray& arr)
 {
+  rclcpp::Clock ros_clock;
+  rclcpp::Time ros_now = ros_clock.now();
```
Comment (ahcorde) [inline moveit_core/collision_distance_field/src/collision_distance_field_types.cpp]: same idea here:

```cpp
arrow_mark.header.stamp = ros_clock.now();
```

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 5]
```diff
@@ -543,25 +546,27 @@ void collision_detection::getProximityGradientMarkers(
...
   {
     for (unsigned int j = 0; j < gradients[i].types.size(); j++)
     {
-      visualization_msgs::Marker sphere_mark;
-      sphere_mark.type = visualization_msgs::Marker::SPHERE;
+      visualization_msgs::msg::Marker sphere_mark;
+      sphere_mark.type = visualization_msgs::msg::Marker::SPHERE;
       sphere_mark.header.frame_id = frame_id;
-      sphere_mark.header.stamp = ros::Time::now();
+      sphere_mark.header.stamp = ros_now;
```
Comment (ahcorde) [inline moveit_core/collision_distance_field/src/collision_distance_field_types.cpp]:
```
sphere_mark.header.stamp = ros_clock.now();
```

**Comment by anasarrak** [comment]
This https://github.com/AcutronicRobotics/geometric_shapes/pull/2 has to be accepted, the CI will fail

File: moveit_core/collision_distance_field/CMakeLists.txt (hunk 1 above)
Comment (anasarrak) [reply moveit_core/collision_distance_field/CMakeLists.txt:33]: On this case I prefer to add a TODO, more than removing it entirely

**Comment by anasarrak** [comment]
Changes made

**Review by ahcorde — APPROVED** [review]

**Pull request merged**